TWE-667 | Blog page updates

kevinhowbrook opened the pull request:
[Link to Ticket](https://torchbox.atlassian.net/browse/TWE-667)

[Link to design](https://www.figma.com/proto/QdMXBHr57mN1svsaNQ7Ggj/Torchbox-2024-2025--Master-?page-id=13440%3A34465&node-id=13440-41955&viewport=37%2C2712%2C0.53&t=oD7wbd4AixE5iSSA-1&scaling=min-zoom&content-scaling=fixed)

### Description of Changes Made

This PR introduces 2 new fields to the blog index page:
- Page links
- Introduction

It also adds functionality to hide filters when there is an intro and/or page links.

> [!NOTE]
> This is BE only, I added some FE but couldn't quite stack the elements correctls


### How to Test

1. Checkout out this branch
2. Run migrations
3. Edit a blog index page
4. Add an introduction value in the new introduction field
5. Add page links via the new streamfield field `page links1
6. These should render on the page above the filters on the listing page
7. Note these aren't required and the page should render nothing when they aren't filled in.

### Screenshots

<details>
  <summary>Expand to see more</summary>
<img width="904" height="817" alt="image" src="https://github.com/user-attachments/assets/2c1b84b4-ba5e-4d1a-9dd0-c4b41f3d7b37" />


<img width="990" height="490" alt="image" src="https://github.com/user-attachments/assets/f207c4e9-27e8-430d-850e-12700fcb0bde" />

</details>

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [x] Updated field spec (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [x] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Light and dark mode

- [ ] I have tested the changes in both light and dark mode
- [ ] The change is not relevant to dark and light mode

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Performance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [x] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [x] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

kevinhowbrook committed:
Adds introduction field for blog index

kevinhowbrook committed:
Add new styled page links blocks for blog index

kevinhowbrook committed:
fixup! Adds introduction field for blog index

zerolab commented on tbx/core/blocks.py:
```diff
@@ -48,6 +48,33 @@ def text(self):
+    class StyleChoice(models.TextChoices):
+        CHARITY = "charity", "Charity"
+        PUBLIC = "public", "Public"
+        WAGTAIL = "wagtail", "Wagtail"
```
suggestion: use the colour theme name a la https://github.com/torchbox/torchbox.com/blob/480360cc67dfbe2e822900ff7ba4fa5e6baa731a/tbx/core/blocks.py#L582-L586 (best to move `ColourTheme` to top level and re-use)

kevinhowbrook committed:
Use existing color theme rather than creating a new one

zerolab commented on tbx/project_styleguide/templates/patterns/pages/blog/blog_listing.html:
```diff
@@ -10,7 +10,8 @@
     <div class="grid grid--spacer-large">
-        {% include "patterns/molecules/title-filters/title-filters.html" with item=page tags=tags %}
+
+        {% include "patterns/molecules/title-filters/title-filters.html" with item=page tags=tags introduction=page.introduction page_links=page.page_links %}
```
oh, yeah. Will have a look after my calls and make some suggestions

ianmeigh replied:
Resolved in 1be005b59e1045492c9e662dcf7222abaa7ab042.

zerolab commented on tbx/blog/migrations/0031_blogindexpage_page_links.py:
```diff
@@ -0,0 +1,21 @@
+# Generated by Django 5.2.5 on 2025-10-22 09:34
```
forgot to add earlier: could you squash these into one? probably easiest is to remoe and regenerate

kevinhowbrook committed:
Squash blog migrations: combine introduction and page_links fields

- Remove separate migration files 0030 and 0031
- Create single migration 0030_blogindexpage_introduction_and_page_links
- Combines both field additions into one migration as requested in PR feedback

shyusu4 committed:
Update styled page link block FE: update class names and clean up styles

shyusu4 committed:
Add introductiona and styles page links FE

kevinhowbrook commented:
> Hey @kevinhowbrook do you have what you need now from @shyusu4 ? Let me know if you need code review.

Thanks @helenb - with @shyusu4 and @zerolab 's input I think this is ready for review now yea.

helenb reviewed (commented):
Hey @kevinhowbrook and @shyusu4 sorry it's taken so long to review this. I spotted just one small issue with the pattern library but otherwise this looks good.

ianmeigh committed:
refactor: correct relative paths in Sass @use statements

ianmeigh committed:
fix: reduce width of before content in grid layout

Fix to address visual inconsistency between blog pages with tags and others with an introduction. We currently use flex when displaying tags, and grid when displaying information elements.

ianmeigh committed:
chore: fix pattern library style for styled_page_link_block

ianmeigh commented on tbx/project_styleguide/templates/patterns/molecules/streamfield/blocks/styled_page_link_block.yaml:
```diff
@@ -0,0 +1,12 @@
+      title: Charity
+      url: '#'
+    title: Charity
+    style: charity
```
Updated to `coral` in 8ce62da1223c9e1319ddd10375b08ebe86282cbf.

ianmeigh committed:
Revert "fix: reduce width of before content in grid layout"

This reverts commit b83711d986cbd1d11fae84f576e325939a609a49.
